goodtablesio/models/job.py

```diff
@@ -110,15 +113,53 @@ def get_ids():
     return [j.id for j in job_ids]
 
 
-def get_all():
-    """Get all jobs in the database as dict.
+def find(filters=None, limit=10, offset=0):
```

Comment:
Whenever there is a query with a limit that can be set as a param there should be a hardcoded limit. Another question is if the helper should be in models at all :) 

> it's way more additional code to write and test

I still think that yes, it makes sense to push as some stuff that interacts with the database to models, specially something like getting a list of jobs which we will do all over the place (`find()` is already called from 4 different modules). I don't see like having blueprints, tasks, tests, etc having to duplicate the same queries, session management etc would improve the code. Or when having to update a job on the DB from a task we should have to worry about writing the whole query, commiting etc instead of just doing `models.job.update({'id': 'xxx', 'status': 'running'})`. It might seem like a lot of code but if we multiply this to all places where the helper is used I think it's well worth the duplication we are avoiding

> it's not enough flexible - in many place we need real model instances, special queries, ability to control transactions (as in sync_user_repos task) etc.

I agree completely that this will probably won't be flexible enough and all cases, then you just wrap ORM queries from your blueprint / task / whatever (I wouldn't add `models.job.get_last_jobs_per_github_organization`, but I would put in `plugins.github.models`). Perhaps some of the current helpers in models won't be needed after all, we can clean up later.

One thing I'm happy to consider is returning ORM objects rather than dicts from the model layer (not calling `.to_dict()`) and leaving this to view functions etc. This would allow to keep using further SA logic if you need it. I think this is closer to your concerns than having the helpers in `models` so definitely open to change it.

> ORM API is a standard way to interact with models (so all developers are familiar with it) - no need for additional level of abstraction

I agree, but it also a good practice to separate concerns. Maybe I'm biased from much bigger applications like CKAN but I always felt more comfortable with views handling requests, models interacting with DB, etc